### PR TITLE
OPSEXP-2333 Fix the issue with the ImageMagick packages for RockyLinux 9

### PR DESCRIPTION
**Pull request by slohe1**
OPSEXP-2333

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -69,6 +69,69 @@ jobs:
           maven-username: ${{ secrets.NEXUS_USERNAME }}
           maven-password: ${{ secrets.NEXUS_PASSWORD }}
 
+  test_and_run_command:
```
Comment (alxgomz) [inline .github/workflows/build.yml:72]: this job is a condition to deploy top nexus. We don't want to deploy on nexus if the tests fail

File: rpms/after-checkout-rockylinux9-7.1.0-16.sh
Comment (alxgomz) [inline rpms/after-checkout-rockylinux9-7.1.0-16.sh:1]: That's ^pretty difficult to read :)
Can you explain what was the problem?

**Review by alxgomz — CHANGES_REQUESTED** [review]
Ideally we should test `.deb`as well. If you feel like it's fast enough please implement, otherwise create a new JIRA for that

**Review by gionn — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -6,24 +6,27 @@ on:
...
+    steps:
+     - name: Checkout to repository
+       uses: actions/checkout@v3
+     - name: Set matrix data
+       id: set-matrix
+       run: |
+         echo "matrix=$(jq -c . < ./config.json)" >> $GITHUB_OUTPUT
+         echo "matrix_test=$(jq -c 'del(.include[] | select(.base_image == "rockylinux:8" and .target_arch == "aarch64" and .nexus_classifier == "el8-aarch64"))' < config.json)" >> $GITHUB_OUTPUT
+
+  Build:
```
Comment (gionn) [inline .github/workflows/build.yml:23]: lowercase

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -52,6 +55,58 @@ jobs:
         with:
           name: ${{ matrix.nexus_classifier }}-rpm
           path: RPMS/${{ matrix.target_arch }}/*.rpm
+
+  Test:
```
Comment (gionn) [inline .github/workflows/build.yml:59]: lowercase

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -52,6 +55,58 @@ jobs:
...
+    name: Test on ${{ matrix.base_image }} for ${{ matrix.target_arch }}
+    runs-on: ubuntu-latest
+    needs: [configure,Build]
```
Comment (gionn) [inline .github/workflows/build.yml:62]: alway use the multiline notation for readability

File: config.json
Comment (gionn) [inline config.json:1]: move it under rpms folder to make it clear that is rpm-only

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -52,6 +55,58 @@ jobs:
...
+          echo "ldd output of convert"
+          docker exec -t testenv ldd /usr/bin/convert
+          echo "Testing convert command"
+          docker exec -t testenv convert -version
+          echo "Creating test image file"
+          convert -size 32x32 xc:transparent test.png
+          echo "Converting png to jpg"
+          convert test.png test1.jpg
+
       - name: Deploy to nexus
```
Comment (gionn) [inline .github/workflows/build.yml:110]: at this point I prefer having a dedicated deploy job - feels a bit cheating that we are doing deploy at the end of a job which is called 'test'

**Review by gionn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**